### PR TITLE
Add optional third party customization support

### PR DESCRIPTION
**Pull request by bernie-laberge**
### Linked issues
NA

### Summarize your change.

In this commit:

- Added RV_INTERNAL_ORGANIZATION_NAME + RV_INTERNAL_ORGANIZATION_DOMAIN cmake cached variable for additional customization so the RV Prefs can be saved in a different organization's name than Autodesk.

- Added some optional cmake global properties that can be set to perform some custom RV customizations for superprojects consuming OpenRV via a submodule for example.

- The optional cmake global properties that were added are:

RV_THIRD_PARTY_CUSTOMIZATION (rv main)
RVIO_THIRD_PARTY_CUSTOMIZATION (rvio main)
RV_VIEW_GROUP_THIRD_PARTY_CUSTOMIZATION (RvViewGroupIPNode::evaluate())
RV_LAZY_BUILD_THIRD_PARTY_CUSTOMIZATION (RvApplication::lazyBuild())

To leverage this extra customization in your super project, all is required to do is to set the cmake global property to a target of your super project handling this third party customization.
Example:

SET_PROPERTY(
    GLOBAL
    PROPERTY "RV_THIRD_PARTY_CUSTOMIZATION" rvThirdPartyCustomization
)
Where rvThirdPartyCustomization is a target from your super project implementing void rvThirdPartyCustomization(...)

Also in this commits:

- Removed some dead code.

### Describe the reason for the change.

This optional generic third party customization was required to rebase the commercial version of RV on top of Open RV where Open RV is consumed as a sub-module.

### Describe what you have tested and on which operating system.

Tested on Mac. Will also validate on other platforms.

### If possible, provide screenshots.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,6 +54,14 @@ SET(RV_INTERNAL_APPLICATION_NAME
     "OpenRV"
     CACHE STRING "RV's internal application name"
 )
+SET(RV_INTERNAL_ORGANIZATION_NAME
+    "Autodesk"
```
Comment (geffrak) [inline CMakeLists.txt:58]: Should it be Autodesk? It was Tweak before. Perhaps it should be something about the Academy Software Foundation?

**Review by geffrak — APPROVED** [review]
Thanks!

File: CMakeLists.txt (hunk 1 above)
Comment (bernie-laberge) [reply CMakeLists.txt:58]: True ! How about ASWF ?

Comment (geffrak) [reply CMakeLists.txt:58]: Looks good to me.

Comment (bernie-laberge) [reply CMakeLists.txt:58]: Alain proposes OpenRV as the organisation's name. Any objection ?

Comment (geffrak) [reply CMakeLists.txt:58]: Same application name and organization name? It's weird but not impossible.

Comment (bernie-laberge) [reply CMakeLists.txt:58]: OK Alain told me to go with ASWF.

Comment (bernie-laberge) [reply CMakeLists.txt:58]: I'll make the change. Good catch Kerby ! Thanks !

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,6 +54,14 @@ SET(RV_INTERNAL_APPLICATION_NAME
...
+    CACHE STRING "RV's internal organization name"
+)
+SET(RV_INTERNAL_ORGANIZATION_DOMAIN
+    "autodesk.com"
```
[the remark this answers is not in the document]
Comment (bernie-laberge) [reply CMakeLists.txt:62]: Yes of course : I am currently testing aswf.io as the organization's domain name but it doesn't seem to behave the way I was expecting it to (even after adapting the code in PackageManager.cpp (defaultSettingsFileName()) which was assuming ".com" 
I am currently rebuilding from scratch to investigate this. I'll report back once I've been to the bottom of it.

**Pull request merged**